.github/workflows/cicd-develop-deploy.yaml

```diff
@@ -0,0 +1,80 @@
+name: CI/CD Develop Pipeline
+
+on:
+  push:
+    branches:
+      - main
+      - 'task/*'
+
+jobs:
+  get-metadata:
+    runs-on: ubuntu-latest
+    outputs:
+      build_datetime: ${{ steps.metadata.outputs.build_datetime }}
+      build_timestamp: ${{ steps.metadata.outputs.build_timestamp }}
+      build_epoch: ${{ steps.metadata.outputs.build_epoch }}
+    steps:
+      - id: metadata
+        run: |
+          datetime=$(date -u +'%Y-%m-%dT%H:%M:%S%z')
+          echo "build_datetime=$datetime" >> $GITHUB_OUTPUT
+          echo "build_timestamp=$(date --date=$datetime -u +'%Y%m%d%H%M%S')" >> $GITHUB_OUTPUT
+          echo "build_epoch=$(date --date=$datetime -u +'%s')" >> $GITHUB_OUTPUT
+          echo "triggering action: ${{ github.event_name }}"
+          echo "ref type: ${{ github.ref_type }}"
+          echo "triggering reference: ${{ github.ref_name }}"
+  scan-secrets:
+    uses: ./.github/workflows/scan-secrets.yaml
+  check-file-format:
+    uses: ./.github/workflows/check-file-format.yaml
+  check-markdown-format:
+    uses: ./.github/workflows/check-markdown-format.yaml
+  check-terraform-format:
+      uses: ./.github/workflows/check-terraform-format.yaml
+  validate-terraform:
+    name: Validate terraform code
+    uses: ./.github/workflows/validate-terraform.yaml
+    with:
+      ENV: dev
+    secrets: inherit
+  check-python-black-format:
+      uses: ./.github/workflows/check-python-black-format.yaml
+  check-python-flake-rules:
+      uses: ./.github/workflows/check-python-flake-rules.yaml
+  cloc-repository:
+    uses: ./.github/workflows/cloc-repository.yaml
+  cicd-pipeline-test:
```

Comment:
I know we are thinking of moving the format/code qualty checks to a separate workflow and that this is 'testing" the quality and format ..but we have a placeholder  job with same name in the test-deploy workflow that is designed to run tests. We would also presumably be dropping in a unit-test job - so is it worth renaming this one to make a clear distinction between it and the test one